.github/workflows/ci_tests_dev.yaml

```diff
@@ -0,0 +1,79 @@
+# This workflow installs PyGMT dependencies, builds documentation and runs tests on GMT master
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-python-with-github-actions
+
+name: GMT Master Tests
+
+on:
+  # push:
+  #   branches: [ master ]
+  pull_request:
+    types: [review_requested, ready_for_review]
```

Comment:
Found this nice way to limit the GMT Master Tests to non-draft PRs (see https://github.community/t/dont-run-actions-on-draft-pull-requests/16817/6 and also https://docs.github.com/en/actions/reference/events-that-trigger-workflows#pull_request). I think this strikes a nice balance between testing GMT master "on every PR commit" and testing "only on the PyGMT master branch". Let me know if this is okay.

As an aside, we could apply this to other expensive tests too (e.g. \**coughs*\* Windows).

Reply:
Looks a good idea. Do we still want to test GMT master when a PR is merged into master branch?

Reply:
> Do we still want to test GMT master when a PR is merged into master branch?

Up to you. I'd prefer to save on CI resources, and there's a daily cron job anyway.

Reply:
OK. Daily cron job is enough.

Reply:
OK. Daily cron job is enough.

Reply:
I just made a new commit b91136697dfd0abb836f0593c332e16a8b3911f3 to this branch. The GMT master workflow is not triggered. I need to click the "Re-request review" icon to trigger the GMT master test.

I think It's good. The GMT master workflow only runs when:

1. A draft PR is ready for review
2. A review is request or re-request

Reply:
Yes, it's a nice feature. We don't need to test this too often anyway so I'm happy with it :smile: Should I document this in MAINTENANCE.md?

Reply:
Yes, please.